packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2896,7 +2897,11 @@ function commitPassiveMountOnFiber(
                 // Add all the transitions saved in the update queue during
                 // the render phase (ie the transitions associated with this boundary)
                 // into the transitions set.
-                prevTransitions.add(transition);
+                if (prevTransitions === null) {
```

Comment:
PrevTransitions can't be null because we set it above to a new set if it is 

```
if (transitions !== null && prevTransitions === null) {
    instance.transitions = prevTransitions = new Set();
}
```